internal/api/router.go

```diff
@@ -18,10 +20,11 @@ type Service interface {
 }
 
 // New returns an 'http.Handler' that serves the ship-it API.
-func New(s Service) http.Handler {
+func New(s *service.Service) http.Handler {
```

Comment:
The histogram should be provided as a dependency to the API handler, since that's where it's used, rather than the service.